Major changes in weekly release line (#3943)
jenkins-infra/jenkins.io: content/blog/2020/11/2020-11-08-help-test-jenkins.adoc

---
layout: post

> title: "Help Us Test Jenkins"

Reviewer: ```suggestion
title: "Jenkins 2.264+: Major changes in the weekly release line"
```

tags:

> - jenkins

Reviewer: ```suggestion
- jenkins
- core
```

- testing
- ui
author: markewaite

> description: "Latest Jenkins weekly releases contain large changes.  Help us test the most recent releases"

Reviewer: ```suggestion
description: "Latest Jenkins weekly releases contain large changes. We invite Jenkins users to evaluate these changes and provide their feedback"
```

opengraph:

>   image: /images/post-images/2020-11-08-help-test-jenkins/help-test-jenkins.png

Reviewer: Two-level list does not look well IMHO, I would remove UI/Core separation.

Author: Revised in 701c32ef454ad17e3e637001c69db7483c8ff625

Reviewer: Thanks

---

> Beginning with the Jenkins 2.264 release on October 27, 2020, we've entered a period where the Jenkins weekly releases will include more significant changes than usual.

Reviewer: ```suggestion
Recently we have link:https://groups.google.com/g/jenkinsci-dev/c/AWOPeU4k0DI[selected Jenkins 2.263 as a new baseline] for the link:/download/lts/[LTS release line], with ETA in December 2021.
It allows delivering significant and in some cases breaking changes which have been previously on hold.
Beginning with the Jenkins 2.264 release on October 27, 2020, we've entered a period where the Jenkins weekly releases will include more significant changes than usual.
```

> That period of more significant changes is expected to continue for a month or more.

Reviewer: ```suggestion
That period of more significant changes is expected to continue for a month or more.
As you may have seen from the link:/changelog/[release community ratings], there might be regressions and instabilities during this period.
```

> We're excited for the changes and invite you to join us as we test those changes.

Reviewer: ```suggestion
We're excited for the changes.
They help to improve user experience and to address the technical debt accumulated in the Jenkins core.
We invite Jenkins users to evaluate those changes and provide feedback.
```

> This is an especially valuable time for users and administrators to test the weekly releases and report issues with them.

Reviewer: ```suggestion
This is an especially valuable time for users and administrators to test the weekly releases and report issues with them,
especially on Jenkins test environments.
In the Jenkins project we have invested a lot in test coverage for the main functionality,
but in many cases we rely on user feedback for exotic plugins and environments not yet covered by our test automation.
```

> The changes include:

Reviewer: ```suggestion
The most notable changes include:
```

It allows delivering significant and in some cases breaking changes which have been previously on hold.
Beginning with the Jenkins 2.264 release on October 27, 2020, we've entered a period where the Jenkins weekly releases will include more significant changes than usual.
That period of more significant changes is expected to continue for a month or more.
As you may have seen from the link:/changelog/[release community ratings], there might be regressions and instabilities during this period.

We're excited for the changes.
They help to improve user experience and to address the technical debt accumulated in the Jenkins core.
We invite Jenkins users to evaluate those changes and provide feedback.
This is an especially valuable time for users and administrators to test the weekly releases and report issues with them,
especially on Jenkins test environments.
In the Jenkins project we have invested a lot in test coverage for the main functionality,
but in many cases we rely on user feedback for exotic plugins and environments not yet covered by our test automation.

The most notable changes include:

> * <<UI - Tables to Divs>>

Reviewer: ```suggestion
* <<Configuration UI - Tables to Divs>>
```

* <<Core - Spring Security replaces Acegi Security>>
* <<Core - XStream unfork>>
* <<UI - JQuery upgrade>>

> == UI - Tables to Divs

Reviewer: ```suggestion
== Configuration UI - Tables to Divs
```

> Jenkins 2.264 is the first weekly release to include the "Tables to Divs" user interface work of link:https://github.com/jsoref/[Josh Soref], link:https://github.com/timja/[Tim Jacomb], and https://github.com/fqueiruga[Felix Queiruga].

Reviewer: ```suggestion
Jenkins 2.264 is the first weekly release to include the "Tables to Divs migration" user interface work of link:https://github.com/jsoref/[Josh Soref], link:https://github.com/timja/[Tim Jacomb], and https://github.com/fqueiruga[Felix Queiruga].
```

> It is a significant step to improve the Jenkins user interface, especially for users on narrow devices like tablets and phones.

Reviewer: ```suggestion
It is a significant step to improve forms in the Jenkins user interface (configuration pages, build parameters, etc.), especially for users on narrow devices like tablets and phones.
```

If you're not comfortable proposing a pull request, describe the problems you see in a bug report.

A link:/doc/developer/views/table-to-div-migration/[tables to divs migration guide] is available.
It describes areas that typically need to be changed as part of the migration from tables to divs.
It also includes detailed examples that allow the plugin to continue supporting older Jenkins versions with table layouts and use div layouts for newer Jenkins versions.

== Core - Spring Security replaces Acegi Security

The Jenkins 2.266 release on November 9, 2020 will include the migration to the Spring Security libraries from the Acegi security libraries that link:https://github.com/jglick[Jesse Glick] has proposed and developed through jep:227[Jenkins Enhancement Proposal 227].

This upgrade replaces the Acegi Security library with the current release of the Spring Security library.
Details of the change are described in jep:227[] and in the link:https://github.com/jenkinsci/jenkins/pull/4848[pull request].

We need users to test the latest Jenkins weekly releases with their plugins and watch for issues related to authentication.

> Refer to Jesse Glick's blog post that introduces the details of the change and provides links to the link:https://github.com/jenkinsci/jep/blob/master/jep/227/compatibility.adoc[Spring Security compatibility] table.

Reviewer: Not released yet. Are we blocked by it. Maybe just add "coming soon"?

Author: I merged Jesse's blog post so that it would be visible with tomorrow's release of Jenkins 2.266.  We can merge this with the release of 2.266 or wait one day and then merge it.  Either is great with me.

It is a significant step to improve the Jenkins user interface, especially for users on narrow devices like tablets and phones.

=== A better user interface

The transition from using HTML `table` tags to using HTML `div` tags provides a more attractive user interface for all users and a much better experience for users on narrower devices.
Before the conversion from table tags to div tags, the **"Manage Jenkins"** page looked like this in a 1024x768 window:

image:/images/post-images/2020-11-08-help-test-jenkins/ui-using-tables.png[Before the pull request]

After the conversion, the **"Manage Jenkins"** page now looks like this:

image:/images/post-images/2020-11-08-help-test-jenkins/ui-using-divs.png[After the pull request]

The user interface improvements from the transition are a nice step forward for Jenkins.
However, because the user interface improvements require changes in plugins, we need your help.

We need users to test the latest weekly Jenkins releases with the plugins and configurations that are most important to them.

> When users detect an issue, we need them to link:/participate/report-issue/[report the issue] with enough detail that a plugin maintainer can fix the issue.

Reviewer 2: add the 'tables-to-divs-regression' label?

Author: Thanks!  Included in e841b017df31569c6917a11b3eb80d620b740cfc

=== Plugin developers

In addition to testing, we've already identified several plugins that may need changes.

> See the link:https://issues.jenkins-ci.org/browse/JENKINS-62437[Jira epic] for plugins that are likely to need changes for the new user interface layout.

Reviewer 2: these are the plugins that are known to break: https://issues.jenkins-ci.org/browse/JENKINS-64142?filter=22840

in many cases tabular markup still works, but it's good to get rid of it.

Author: Thanks also.  Included in ee8a50a93090e1f505af3cda2bc2179ac15b5cca

> If you can assist with plugin testing and code changes, select one of the plugins from that epic, test it, and propose a pull request to help with this user interface transition.

Reviewer 2: migration guide should be mentioned https://www.jenkins.io/doc/developer/views/table-to-div-migration/

Author: Added a migration guide paragraph in 2aa7e04126470a7464c732fae8af214f5dd3afa2
